pkg/utils/common_test.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestChunks(t *testing.T) {
+	size := 2
+	obj1 := &struct{}{}
+	obj2 := &struct{}{}
+	obj3 := &struct{}{}
+
+	t.Run("[obj1] -> [[obj1]]", func(t *testing.T) {
+
+		in := []*struct{}{obj1}
+		out := Chunks(in, size)
+		assert.Equal(t, [][]*struct{}{{obj1}}, out)
+
+	})
+
+	t.Run("[obj1,obj2] -> [[obj1,obj2]]", func(t *testing.T) {
+
+		in := []*struct{}{obj1, obj2}
+		got := Chunks(in, size)
+		assert.Equal(t, [][]*struct{}{{obj1, obj2}}, got)
+
+	})
+
+	t.Run("[obj1, obj2, obj3] -> [[obj1,obj2],[obj3]]", func(t *testing.T) {
+		in := []*struct{}{obj1, obj2, obj3}
+		got := Chunks(in, size)
+		assert.Equal(t, [][]*struct{}{{obj1, obj2}, {obj3}}, got)
+	})
```

Comment:
Usually these functions tested well with table-style tests. Also sub-tests cover edge cases and normal cases. [obj1] -> [[obj1]] weird name of single item. Need to cover: no-items, single item, multiple items but single partition, mutliple items 2 partitions, multiple items 2+ partitions, partition size of 0(?)